azure-kusto-ingest/source/streamingIngestClient.js

```diff
@@ -28,41 +28,29 @@ module.exports = class KustoStreamingIngestClient {
         return this.defaultProps.merge(newProperties);
     }
     
-    ingestFromStream(stream, ingestionProperties, callback) {
+    async ingestFromStream(stream, ingestionProperties) {
         const props = this._mergeProps(ingestionProperties);
-
-        try {
-            props.validate();
-        } catch (e) {
-            return callback(e);
-        }
+        props.validate();
 
         const descriptor = stream instanceof StreamDescriptor ? stream : new StreamDescriptor(stream);
         const compressedStream  = 
-            descriptor.compressionType == CompressionType.None ? descriptor._stream.pipe(zlib.createGzip()) : descriptor._stream;
+            descriptor.compressionType == CompressionType.None ? descriptor.stream.pipe(zlib.createGzip()) : descriptor.stream;
```

Comment:
Minor + not yours:
Can we please format this (and others, if you feel like it) the same way as in the main repo?
Condition
? term1
: term2

Reply:
we format like this in the main repo? 😮
actually, I think that I prefer the current format...